crates/sovereign-sdk/rollup-interface/src/node/rpc/mod.rs

```diff
@@ -178,7 +178,7 @@ pub struct SoftBatchResponse {
     #[serde(with = "hex::serde")]
     pub hash: [u8; 32],
     /// The transactions in this batch.
-    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(skip_serializing_if = "Option::is_none", with = "utils::tx_hex")]
     pub txs: Option<Vec<Vec<u8>>>,
```

Comment:
So basically txs can be an Option<TxsVec> where TxsVec is a flatten proxy for serde::with hex. This can be reduce our code (not having to implement our custom ser/de visitors is a plus) but can lead to more patches in the code.

Reply:
yeah I agree with @kpp here, 

if we can get away without needing custom serializers 

it's much better 😄

Reply:
@kpp I do not get what you mean by "flatten proxy" here. Regardless of the option, using `#[serde(with = "hex::serde")]` would not work on the inner `Vec<Vec<u8>>` but would work on `Vec<u8>` as `hex::serde::serialize` expects a slice of u8s. 
If you can elaborate further, i will be happy to simplify the code.

Reply:
I mean `#[serde(transparent)]`.